source/dagre.js

```diff
@@ -1740,7 +1740,7 @@ dagre.layout = (graph, layout) => {
                 for (const [v, x] of Object.entries(xs)) {
                     const halfWidth = g.node(v).label.width / 2;
                     max = Math.max(x + halfWidth, max);
-                    min = Math.min(x - halfWidth, min);
+                    min = Math.min(Math.max(0, x - halfWidth), min);
```

Comment:
some `x - halfWidth` goes to minus, which will make below `const width = max - min;` larger.